### ISSUE TITLE
Slow training and validation loss doesn’t decrease when using encoding.nn.SyncBatchNorm

### ISSUE THREAD

**Issue by John1911603424** [zhanghang1989/PyTorch-Encoding#271]
Slow training and validation loss doesn’t decrease when using encoding.nn.SyncBatchNorm

**Comment by John1911603424** [comment]
请问一下，用了您的pytorch版本的SyncBN, 语义分割模型训练和验证时loss下降很慢，在30k iters时训练集和验证集的交叉熵loss分别为0.8和0.82左右，一般情况下loss应该在0.4左右，您知道是啥原因造成的吗？

**Comment by zhanghang1989** [comment]
You can use the distributed version, which is 5x faster, but you also lose around 0.5 mIoU.
```
python train_dist.py --dataset ADE20K --model deeplab --aux --backbone resnest50
```

**Comment by zhanghang1989** [comment]
> 请问一下，用了您的pytorch版本的SyncBN, 语义分割模型训练和验证时loss下降很慢，在30k iters时训练集和验证集的交叉熵loss分别为0.8和0.82左右，一般情况下loss应该在0.4左右，您知道是啥原因造成的吗？

用的是我提供的code吗？我一般只在验证集上比较mIoU

**Comment by ad-min570** [comment]
其实我一点都不懂这些，只是无聊乱整，现在好了，那个手机已经崩溃了，哪里都不对劲了，我自己不动都会自己打字在屏幕上，我不知道是不是你弄的，或者是不是别人，现在这个手机上估计也要有，我好像打字，屏幕上做什么人家都知道，手机闪光灯有时候还会自己亮，什么杀毒软件都不行，有些APP我卸载的是时候提示我这个APP不是在我手机上的，还是在别人手机里的，我问别人，我能认识的人都不懂这个，我的明明是安卓系统，怎么用软件扫描的就是电脑？？说可能是什么证书不对，头都大了，手机都坏一个，问题是我还说不出来原因，我身边朋友都不相信我说的，因为我第一，什么都没有，人家会这些的人又不是闲得发慌来弄我，后面我自己也发现我用百度键盘打字，他自己偶尔也会自己打开网页自己输入网址，好像我自己手机屏幕都不是我的，有点APP当你删掉他它时候才发现是另外的名字，我也是醉了。我和你说这些如果你们知道，就会明白，我不懂所以说的有些乱，其实自己也是想好好学的，但已经不是那个年纪，因为要生活｡◕‿◕和你说这些没有别的意思，如果你们懂我说的话或者如果是你们朋友这些，
，请你转告，谢谢！

说实话，其实我也是真心喜欢弄这些！如果有机会，大家交个朋友！也不知道你能不能收到，我刚看了，你的地址是转跳的！不要再是什么不好的链接就行！😂😂😂





------------------ 原始邮件 ------------------




 
 发件人: Hang Zhang  <notifications@github.com>
 

 
 发送时间: 2020-05-01 23:22:32
 

 
 收件人: zhanghang1989/PyTorch-Encoding  <PyTorch-Encoding@noreply.github.com> 
 

 
 抄送: Subscribed  <subscribed@noreply.github.com> 
 

 
 主题: Re: [zhanghang1989/PyTorch-Encoding] Slow training and validation loss doesn’t decrease when using encoding.nn.SyncBatchNorm (#271)
 





 请问一下，用了您的pytorch版本的SyncBN, 语义分割模型训练和验证时loss下降很慢，在30k iters时训练集和验证集的交叉熵loss分别为0.8和0.82左右，一般情况下loss应该在0.4左右，您知道是啥原因造成的吗？

 用的是我提供的code吗？我一般只在验证集上比较mIoU

**Comment by John1911603424** [comment]
> > 请问一下，用了您的pytorch版本的SyncBN, 语义分割模型训练和验证时loss下降很慢，在30k iters时训练集和验证集的交叉熵loss分别为0.8和0.82左右，一般情况下loss应该在0.4左右，您知道是啥原因造成的吗？
> 
> 用的是我提供的code吗？我一般只在验证集上比较mIoU
我只是用了您的单机多卡同步的SyncBN，按照您的例子在我自己的数据集运行的。我也不知道为啥loss下降这么慢。不使用这个SyncBN，loss下降的快一点，同样训练到50k iters, 不使用SyncBN的验证集实验结果会比使用SyncBN的实验结果高一点，而在训练集则高很多。

**Comment by zhanghang1989** [comment]
如果不是我的code，这个很难说，根据一般经验，使用syncbn，loss会下降的更快。

**Comment by John1911603424** [comment]
> 如果不是我的code，这个很难说，根据一般经验，使用syncbn，loss会下降的更快。

谢谢，那我再看看。
